src/pudl_archiver/depositors/depositor.py

```diff
@@ -246,7 +246,7 @@ async def add_resource(
         return await self._apply_change(change)
 
     async def publish_if_valid(
-        self, run_summary: RunSummary, datapackage_updated: bool
+        self, run_summary: RunSummary, datapackage_updated: bool, auto_publish: bool
     ) -> PublishedDeposition | None:
         """Check that deposition is valid and worth changing, then publish if so."""
         logger.info("Attempting to publish deposition.")
```

Comment:
Can we move this down to just above the last return statement? Otherwise we state we're attempting to publish a deposition before it meets our conditions for publication.